test/integration/configuration/conf/test_auth_source_plugin.py

```diff
@@ -0,0 +1,76 @@
+import json
+import os
+import textwrap
+
+import pytest
+
+from conan.test.utils.file_server import TestFileServer
+from conan.test.utils.test_files import temp_folder
+from conan.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+class TestErrorsAuthSourcePlugin:
+    @pytest.fixture
+    def setup_test_client(self):
+        self.client = TestClient(default_server_user=True, light=True)
+        self.file_server = TestFileServer()
+        self.client.servers["file_server"] = self.file_server
+        self.download_cache_folder = temp_folder()
+        self.client.save_home({"global.conf": "tools.files.download:retry=0"})
+
+        save(os.path.join(self.file_server.store, "myfile.txt"), "Bye, world!")
+
+        conanfile = textwrap.dedent(f"""
+                           from conan import ConanFile
+                           from conan.tools.files import download
+                           class Pkg2(ConanFile):
+                               name = "pkg"
+                               version = "1.0"
+                               def source(self):
+                                   download(self, "{self.file_server.fake_url}/basic-auth/myfile.txt", "myfile.txt")
+                           """)
+        self.client.save({"conanfile.py": conanfile})
+
+        return self.client, self.file_server.fake_url
+
+    """ when the plugin fails, we want a clear message and a helpful trace
+    """
```

Comment:
I'll put a minimal docstring to explain a little bit about each test to answer two questions:

* What are we testing?
* What do we expect?